lib/rack/cache/storage.rb

```diff
@@ -38,6 +38,9 @@ def create_store(type, uri)
         else
           fail "Unknown storage provider: #{uri.to_s}"
         end
+      elsif uri.nil?
+        # hack to support setting entitystore to nil, which doesn't persist request bodies
+        type.const_get('NIL').resolve(uri)
```

Comment:
const_get + user-supplied.inspect smells like trouble ...  know it's safe,
but just looks weird :D

On Tue, Jan 26, 2016 at 4:19 PM, Alfredo Amatriain <notifications@github.com
> wrote:

> In lib/rack/cache/storage.rb
> <https://github.com/rtomayko/rack-cache/pull/130#discussion_r50924182>:
>
> > @@ -38,6 +38,9 @@ def create_store(type, uri)
> >          else
> >            fail "Unknown storage provider: #{uri.to_s}"
> >          end
> > +      elsif uri.nil?
> > +        # hack to support setting entitystore to nil, which doesn't persist request bodies
> > +        type.const_get('NIL').resolve(uri)
>
> All other subclasses of EntityStore are instanced like this, calling
> Foo.resolve(uri) which returns the new instance. There is no call to
> Foo.new to instance an entity store in Storage.create_store. I just
> followed the same style.
>
> If what you dislike is the use if the magic string 'NIL' I can change it
> to this:
>
> elsif uri.nil?
>         # hack to support setting entitystore to nil, which doesn't persist request bodies
>         type.const_get(uri.inspect.upcase).resolve(uri)
>
> Better?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rtomayko/rack-cache/pull/130/files#r50924182>.
>